src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensus.scala

```diff
@@ -0,0 +1,569 @@
+package org.constellation.domain.healthcheck
+
+import cats.Order
+import cats.data.NonEmptySet
+import cats.effect.{Blocker, Clock, Concurrent, ContextShift, Fiber}
+import cats.effect.concurrent.Ref
+import cats.syntax.all._
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.circe.{Codec, Decoder, Encoder}
+import io.circe.generic.semiauto.deriveCodec
+import io.circe.syntax.EncoderOps
+import org.constellation.ConstellationExecutionContext.createSemaphore
+import org.constellation.domain.healthcheck.HealthCheckConsensusManager.{
+  FetchProposalError,
+  FetchingFailed,
+  HistoricalRound,
+  SendProposalError,
+  SendingFailed
+}
+import org.constellation.domain.p2p.PeerHealthCheck.{
+  PeerAvailable,
+  PeerHealthCheckStatus,
+  PeerUnresponsive,
+  UnknownPeer
+}
+import org.constellation.infrastructure.p2p.{ClientInterpreter, PeerResponse}
+import org.constellation.p2p.PeerData
+import org.constellation.schema.{Id, NodeState}
+import org.constellation.schema.consensus.RoundId
+
+import scala.concurrent.duration.SECONDS
+import scala.util.{Random, Try}
+
+class HealthCheckConsensus[F[_]](
+  checkedId: Id,
+  ownId: Id,
+  roundId: RoundId,
+  initialRoundPeers: Map[Id, PeerData],
+  healthCheckStatus: PeerHealthCheckStatus,
+  startedAtSecond: Long,
+  val isKeepUpRound: Boolean
+)(
+  apiClient: ClientInterpreter[F],
+  unboundedBlocker: Blocker
```

Comment:
why this is provided separately?

Reply:
This meaning what?

Reply:
```
class Foo(a, b, c)(d, e)(implicit f, g)
```
that (d, e) can be provided altogether with (a, b, c). especially that all of these are ready and you are not using currying for that

Reply:
Ahh, ok it is probably because I thought about separating, data parameters from infrastructure needed to run the logic. Though I don't think I used this approach consistently in other places.